packet_unpacker.go

```diff
@@ -63,43 +63,50 @@ func (u *packetUnpacker) Unpack(headerBinary []byte, hdr *wire.Header, data []by
 			if err != nil {
 				err = qerr.Error(qerr.InvalidAckData, err.Error())
 			}
-		} else {
-			switch typeByte {
-			case 0x01:
-				frame, err = wire.ParseRstStreamFrame(r, u.version)
-				if err != nil {
-					err = qerr.Error(qerr.InvalidRstStreamData, err.Error())
-				}
-			case 0x02:
-				frame, err = wire.ParseConnectionCloseFrame(r, u.version)
-				if err != nil {
-					err = qerr.Error(qerr.InvalidConnectionCloseData, err.Error())
-				}
-			case 0x03:
-				frame, err = wire.ParseGoawayFrame(r, u.version)
-				if err != nil {
-					err = qerr.Error(qerr.InvalidGoawayData, err.Error())
-				}
-			case 0x04:
-				frame, err = wire.ParseWindowUpdateFrame(r, u.version)
-				if err != nil {
-					err = qerr.Error(qerr.InvalidWindowUpdateData, err.Error())
-				}
-			case 0x05:
-				frame, err = wire.ParseBlockedFrame(r, u.version)
-				if err != nil {
-					err = qerr.Error(qerr.InvalidBlockedData, err.Error())
-				}
-			case 0x06:
-				frame, err = wire.ParseStopWaitingFrame(r, hdr.PacketNumber, hdr.PacketNumberLen, u.version)
-				if err != nil {
-					err = qerr.Error(qerr.InvalidStopWaitingData, err.Error())
-				}
-			case 0x07:
-				frame, err = wire.ParsePingFrame(r, u.version)
-			default:
-				err = qerr.Error(qerr.InvalidFrameData, fmt.Sprintf("unknown type byte 0x%x", typeByte))
+		} else if typeByte == 0x01 {
```

Comment:
I think it would be nicer here to duplicate code and have two separate switches for the two cases instead of this if-else mess. WDYT?

Reply:
Also, this could probably be made shorter by combining the err check with the assignments.

Reply:
But there are more than two cases. Not all IETF frames will be introduced at the same time.